runtimes/protocol/lsp.ts

```diff
@@ -40,6 +41,41 @@ export class AutoParameterStructuresProtocolRequestType<P, R, PR, E, RO>
     }
 }
 
+/**
+ * Extra Client information passed during initialization handshake from Client to Server.
+ */
+export interface AWSClientInitializationOptions {
+    /**
+     * Custom client branding information.
+     */
+    product?: {
+        name: string
```

Comment:
Are spaces allowed, or are there any restrictions on the field values?
If so, can these be stated in the comments